packages/testcontainers/src/containers/NativeChainContainer.ts

```diff
@@ -0,0 +1,250 @@
+
+  public withPreconfiguredRegtestMasternode (masterNodeKey = RegTestFoundationKeys[0]): this {
+    this.withMasterNodeKey(masterNodeKey)
+    this.withBlockchainNetwork('regtest')
+    this.hasPreconfiguredMasternode = true
+    return this
+  }
+
+  public static readonly defaultCmd = {
+    prepend: [
+      'defid',
+      '-printtoconsole',
+      '-rpcallowip=0.0.0.0/0',
+      '-rpcbind=0.0.0.0',
+      '-rpcworkqueue=512'
+    ],
+    testnet: [
+      '-testnet=1'
+    ],
+    regtest: [
+      '-regtest=1',
+      '-jellyfish_regtest=1',
+      '-txnotokens=0',
+      '-logtimemicros',
+      '-txindex=1',
+      '-acindex=1',
+      '-amkheight=0',
+      '-bayfrontheight=1',
+      '-bayfrontgardensheight=2',
+      '-clarkequayheight=3',
+      '-dakotaheight=4',
+      '-dakotacrescentheight=5',
+      '-eunosheight=6',
+      '-eunospayaheight=7',
+      '-fortcanningheight=8',
+      '-fortcanningmuseumheight=9',
+      '-fortcanninghillheight=10',
+      '-fortcanningroadheight=11',
+      '-fortcanningcrunchheight=12',
+      '-fortcanningspringheight=13',
+      '-fortcanninggreatworldheight=14',
+      '-fortcanningepilogueheight=15',
+      '-grandcentralheight=16',
+      '-regtest-skip-loan-collateral-validation',
+      '-regtest-minttoken-simulate-mainnet=0'
+    ],
+    mainnet: [],
+    devnet: [],
+    masternode: [
+      '-dummypos=0',
+      '-spv=1',
+      '-anchorquorum=2'
+    ]
+  }
+
+  /**
+   * Convenience Cmd builder
+   */
+  protected generateCmd (): string[] {
+    const { defaultCmd } = NativeChainContainer
+
+    const authCmd = [
+      `-rpcuser=${this.rpcUser}`,
+      `-rpcpassword=${this.rpcPassword}`
+    ]
+
+    const masterNodeOperator = (this.masterNodeKey != null)
+      ? [`-masternode_operator=${this.masterNodeKey.operator.address}`]
+      : []
+
+    return [
+      ...defaultCmd.prepend,
+      ...authCmd,
+      ...defaultCmd[this.blockchainNetwork.name],
+      ...this.hasPreconfiguredMasternode ? defaultCmd.masternode : [],
+      ...masterNodeOperator,
+      ...this.addedCmds
+    ]
+  }
+
+  private readonly addedCmds: string[] = []
+  public addCmd (newCmd: string): this {
+    this.addedCmds.push(newCmd)
+    return this
+  }
+
+  /**
+   * Create container and start it immediately waiting for NativeChain to be ready
+   */
+  public async start (): Promise<StartedNativeChainContainer> {
+    this.withExposedPorts(...(this.hasExposedPorts ? this.ports : Object.values(this.blockchainNetwork.ports)))
+      .withName(this.name ?? this.generateName())
+      .withCommand(this.command.length > 0 ? this.command.concat(this.addedCmds) : this.generateCmd())
+
+    const {
+      rpcUser,
+      rpcPassword,
+      blockchainNetwork,
+      masterNodeKey
+    } = this
+
+    const sncc = new StartedNativeChainContainer(
+      await super.start(),
+      {
+        rpcUser,
+        rpcPassword,
+        blockchainNetwork
+      }
+    )
```

Comment:
```suggestion
    const config = {
      rpcUser: this.rpcUser,
      rpcPassword: this.rpcPassword,
      blockchainNetwork: this.blockchainNetwork,
    }

    const sncc = new StartedNativeChainContainer(await super.start(), config)
```

I think this is more concise and better for readability.